### ISSUE TITLE
preventOverflow.boundariesElement modifier not working properly for scrollParent

### ISSUE THREAD

**Issue by greg-a-smith** [souporserious/react-popper#72]
I'm not sure if this is related to react-popper or the underlying popper.js, but it appears as though the modifier `preventOverflow.boundariesElement` does not work as expected with the value `scrollParent`.  I am testing with a target inside of a scrolling div (inside a modal) and I am not seeing the same behavior as I do when I set `preventOverflow.boundariesElement` to `viewport`.  I am most interested in implementing the "hide" functionality when a target scrolls out of view and I am able to get that working with `viewport`, but **_NOT_** with `scrollParent`.

**Comment by souporserious** [comment]
Sorry for the late reply, is it possible you can provide a demo using something like [codesandbox](codesandbox.io). This sounds like it may be related to PopperJS core.

**Comment by greg-a-smith** [comment]
So this one was interesting.  I stumbled onto the reason this was occurring while attempting to create a branch to demonstrate the issue.  In the end, it appears to be a bug with NPM.

In our project, we have `react-popper` as a dependency defined as `^0.7.4`.  This, in turn, has a dependency on `popper.js` defined as `^1.12.5`.  When I run `npm install` on our project, it ultimately ends up loading `popper.js@1.12.6`.  While attempting to create a branch on `react-popper` demonstrating the issue, I found that running `npm install` was loading `popper.js@1.12.9`.  Why the difference?

The issue with using `scrollParent` for `preventOverflow.boundariesElement` was fixed in `popper.js` in version 1.12.7 with this commit (https://github.com/FezVrasta/popper.js/commit/904d46ba66c285c83d636b56f9566e4318f1fbb3).

I was able to "fix" the issue in our project by adding a top-level dependency for `popper.js` for `^1.12.9` even though we don't directly use `popper.js` in our code.  When I do this, `npm install` ends up loading `popper.js@1.12.9` for both the root dependency _and_ the dependency for `react-popper`.  The `popper.js` under `react-popper` is marked as "deduped".

I created an issue for NPM (https://github.com/npm/npm/issues/19638).  I will close the issue for `react-popper`.
